src/main/java/fi/aalto/cs/apluscourses/intellij/actions/OpenItemAction.java

```diff
@@ -58,14 +61,24 @@ public void actionPerformed(@NotNull AnActionEvent e) {
       return;
     }
 
-    SubmissionResultViewModel submission =
-        (SubmissionResultViewModel) exercisesTree.findSelected().getLevel(3);
-    if (submission == null) {
+    SelectableNodeViewModel<?> nodeViewModel = exercisesTree.getSelectedItem();
+    if (nodeViewModel == null) {
+      return;
+    }
+
+    String url;
+    if (nodeViewModel instanceof SubmissionResultViewModel) {
+      url = ((SubmissionResultViewModel) nodeViewModel).getModel().getHtmlUrl();
+    } else if (nodeViewModel instanceof ExerciseViewModel) {
+      url = ((ExerciseViewModel) nodeViewModel).getModel().getHtmlUrl();
+    } else if (nodeViewModel instanceof ExerciseGroupViewModel) {
+      url = ((ExerciseGroupViewModel) nodeViewModel).getModel().getHtmlUrl();
+    } else {
```

Comment:
As much as I'm allergic to inheritance, a chain like this basically screams for 
```
interface Browsable {
  @NotNull
  String getHtmlUrl();
}
```
which would then be implemented by `ExerciseGroup`, `Exercise`, and `SubmissionResult` (and add `@Override` to the `getHtmlUrl` methods)